packages/engine/src/xr/functions/WebXRFunctions.ts

```diff
@@ -109,13 +109,15 @@ export const proxifyXRInputs = (entity: Entity, inputData: XRInputSourceComponen
  */
 
 export const startWebXR = async (): Promise<void> => {
+  const world = useWorld()
+  if (hasComponent(world.localClientEntity, XRInputSourceComponent)) return
```

Comment:
I think this should probably be handled in upper calling layers (e.g `startXRSession`)  if redundant function calls are concerned. There are more function calls there that should probably not be fired twice per session.